packages/compass-indexes/src/modules/create-index.tsx

```diff
@@ -314,6 +319,19 @@ export type State = {
 
   // base query to be used for query flow index creation
   query: Document | null;
+
+  // to determine whether there has been new query changes since user last pressed the button
+  hasQueryChanges: boolean;
+
+  // covered queries obj for the index flow
+  coveredQueriesObj: {
+    coveredQueries: JSX.Element | null;
+    optimalQueries: string | JSX.Element | null;
```

Comment:
JSX elements shouldn't be in the state, the purpose of the state is to keep the minimal, serializeable, source-of-truth values, how exactly they are rendered is a derived state